### DISCUSSION TITLE
Websocket connection to `relay.walletconnect.com` fails -> WalletConnect v2 modal doen't open

### DISCUSSION THREAD

**Discussion by michaelsmueller** [WalletConnect/walletconnect-monorepo#3835] [Developer Support] [answered]
## Issue

We are experiencing an issue in our test environment where the websocket connection to the WalletConnect relay hangs. This issue doesn't occur in our production environment. The request is:
```
wss://relay.walletconnect.com/?auth=...<truncated>...&useOnCloseEvent=true
```

## Behavior

In our test environment, the handshake to WalletConnect fails without any error message. In production, the handshake is successful:

prod (works)
<img width="867" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/3327056/e9dc32ac-a533-428a-a8bb-a80e47634e6a">

test (fails)
<img width="995" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/3327056/903e30d0-55c4-43a2-b1dd-5ec103783e0e">

We have verified our domain in the project configuration on cloud.walletconnect.com and have added our origin to the Allowed Domains. Before this configuration, the wss request used to fail with an unauthorized error message, which is no longer the case.

## Associated bug - modal doesn't open

This issue seems to be causing a bug where the WalletConnect v2 modal fails to render in our React app, with the following error in the console (on load):

```
index-7371c16c.js:35 Error: To use QR modal, please install @walletconnect/modal package
    at k1t.initialize (index-7371c16c.js:2226:43156)
    at async k1t.init (index-7371c16c.js:2226:38473)
```

Note we import the WalletConnect connector from `@web3-react/walletconnect-v2` (which has `@walletconnect/modal` as a dependency), with modal opening triggered when activating the connector (`connector.activate(CHAIN_ID)`).

## Investigation

We believe we've ruled out potential issues relating to:
- CORS
- CSP
- Build / tree shaking (we're using `vite`)

## Request

Guidance on how we might resolve this, e.g. are there any known issues or configurations we might have overlooked?

**Comment by glitch-txs** [comment, marked as answer]
they need to update WalletConnect dependencies, this should be fixed in the latest version

Also for reading calls I would suggest using a custom RPC endpoint instead of the wallet or public provider which is not reliable.

**Reply by michaelsmueller** [reply]
Thanks for the response

For anyone else experiencing this issue, while we wait for `@web3-react` to update their library, fixing the following dependencies in our `package.json` is a fix:
```json
  "resolutions": {
    "@walletconnect/core": "2.10.2",
    "@walletconnect/ethereum-provider": "2.10.2"
  }
```
Issue seems to arise when updating dependencies (e.g. `yarn up`) to `@walletconnect` `2.10.4`.

We also committed the latest `yarn` (v4) to our repo.
